### PR TITLE
Adding tests to avatar deletion flow

### PR DESCRIPTION
**Pull request by etoledom**
Closes #

### Description

Adding unit tests to the Avatar deletion flow.

### Testing Steps

- CI should be 🍏 
- Smoke test Avatar deletion

### CODE REVIEW

File: Sources/GravatarUI/SwiftUI/AvatarPicker/AvatarGridModel.swift [hunk 1]
```diff
@@ -66,8 +66,8 @@ class AvatarGridModel: ObservableObject {
         }
     }
 
-    func deleteModel(_ id: String) -> Int {
-        let index = avatars.firstIndex { $0.id == id }!
+    func deleteModel(_ id: String) -> Int? {
+        guard let index = avatars.firstIndex(where: { $0.id == id }) else { return nil }
         avatars.removeAll { $0.id == id }
```
Comment (etoledom) [inline Sources/GravatarUI/SwiftUI/AvatarPicker/AvatarGridModel.swift:71]: Nasty force unwrap went through. 
It was probably this new TAB autocomplete thing and I didn't notice it 🤦 

Deleting an avatar from the grid can fail if it's not found, in this case the deletion index returned is `nil`.

File: Sources/GravatarUI/SwiftUI/AvatarPicker/AvatarPickerViewModel.swift [hunk 2]
```diff
@@ -368,7 +370,7 @@ class AvatarPickerViewModel: ObservableObject {
             try await avatarService.delete(avatarID: avatar.id, accessToken: token)
             return true
         } catch APIError.responseError(let reason) where reason.httpStatusCode == 404 {
-            return false // no-op. We delete a not-found avatar from the UI.
+            return true // no-op. We delete a not-found avatar from the UI.
         } catch APIError.responseError(reason: let reason) where reason.urlSessionErrorLocalizedDescription != nil {
```
Comment (etoledom) [inline Sources/GravatarUI/SwiftUI/AvatarPicker/AvatarPickerViewModel.swift:376]: Returning `false` on this error could prevent an avatar changed notification. We want the notification as we are effectively removing this avatar from the grid, so we treat it as a success.

File: Sources/GravatarUI/SwiftUI/AvatarPicker/AvatarGridModel.swift (hunk 1 above)
Comment (pinarol) [reply Sources/GravatarUI/SwiftUI/AvatarPicker/AvatarGridModel.swift:71]: Oh no, I had overlooked it as well.

**Review by pinarol — APPROVED** [review]
Awesome! 🎉

**Pull request merged**